docs/source/markdown/podman-run.1.md

```diff
@@ -892,20 +892,26 @@ Specify the policy to select the seccomp profile. If set to *image*, Podman will
 
 Note that this feature is experimental and may change in the future.
 
-#### **\-\-secret**=*secret*
+#### **\-\-secret**=*secret*[**,**opt=opt]
 
 Give the container access to a secret. Can be specified multiple times.
 
 A secret is a blob of sensitive data which a container needs at runtime but
 should not be stored in the image or in source control, such as usernames and passwords,
 TLS certificates and keys, SSH keys or other important generic strings or binary content (up to 500 kb in size).
 
-Secrets are copied and mounted into the container when a container is created. If a secret is deleted using
-`podman secret rm`, the container will still have access to the secret. If a secret is deleted and
-another secret is created with the same name, the secret inside the container will not change; the old
+When secrets are specified as type `mount`, the secrets are copied and mounted into the container when a container is created.
+When secrets are specified as type `env`, the secret will be set as an enviornment variable.
+If a secret is deleted using `podman secret rm`, the container will still have access to the secret.
+If a secret is deleted and another secret is created with the same name, the secret inside the container will not change; the old
 secret value will still remain.
 
-Secrets are managed using the `podman secret` command
+Secrets are managed using the `podman secret` command.
+
+Secret Options
+
+- `type=mount|env`    : How the secret will be exposed to the container. Default mount.
```

Comment:
It feels like this should be decided at secret creation time?

Reply:
I think it makes more sense for it to be an option in container run/create since mount/env doesn't change the substance of the secret, but only how the secret is used in relation to the container. More discussion here: https://github.com/containers/common/pull/516

Reply:
@vrothberg and I had this discussion before and went back and forth on it, we came to this conclusion, to allow users to specify how they want to share the secrets.